fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java

```diff
@@ -24,38 +21,29 @@ public void syncTable(ExternalOlapTable table) {
         String host = table.getSourceTableHost();
         int port = table.getSourceTablePort();
         TNetworkAddress addr = new TNetworkAddress(host, port);
-        FrontendService.Client client = null;
-        try {
-            client = ClientPool.frontendPool.borrowObject(addr, 1000);
-        } catch (Exception e) {
-            LOG.warn("get frontend client from pool failed", e);
-            return;
-        }
-
         TGetTableMetaRequest request = new TGetTableMetaRequest();
         request.setDb_name(table.getSourceTableDbName());
         request.setTable_name(table.getSourceTableName());
         TAuthenticateParams authInfo = new TAuthenticateParams();
         authInfo.setUser(table.getSourceTableUser());
         authInfo.setPasswd(table.getSourceTablePassword());
         request.setAuth_info(authInfo);
-        boolean returnToPool = false;
         try {
-            TGetTableMetaResponse response = client.getTableMeta(request);
-            returnToPool = true;
-            List<String> errmsgs = response.status.getError_msgs(); 
+            TGetTableMetaResponse response = FrontendServiceProxy.call(addr, 10000,
+                    client -> client.getTableMeta(request));
             if (response.status.getStatus_code() != TStatusCode.OK) {
-                LOG.info("errmsg: {}", errmsgs.get(0));
+                String errMsg;
+                if (response.status.getError_msgs() != null) {
```

Comment:
When does the error_msg not been set when StatusCode is not ok?

Reply:
This is a protective code.

Reply:
We can not assume every one will set the error_msgs.